x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/SemanticSearchAction.java

```diff
@@ -194,10 +195,26 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalWriteable(storedFields);
         }
 
-        public String[] getIndices() {
+        @Override
+        public String[] indices() {
             return indices;
         }
 
+        @Override
+        public IndicesOptions indicesOptions() {
+            return SearchRequest.DEFAULT_INDICES_OPTIONS;
+        }
```

Comment:
I think it's a surprise to users that a search like API does not support specifying indices options. That said, I noticed the [kNN search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search-api.html) does not support it either. Though I suspect both kNNSearch and SemanticSearch APIs should support it, I am happy for it to be tackled separately.